### DISCUSSION TITLE
Logarithm of an expression

### DISCUSSION THREAD

**Discussion by Hasupama** [firedrakeproject/firedrake#2792] [Firedrake support] [answered]
Hi 
I am new to FEniCS and need to interpolate  logarithm of an expression. I tried with just a log term and got the below error
 
expression:       N.interpolate( log(10 * exp( 1e5* ( (x[0])**2 + (x[1])**2 ) ) ))

error:     TypeError: log() missing 1 required positional argument: 'msg'

How do I obtain log values of this kind of expression?

**Comment by dham** [comment, marked as answer]
First, this isn't a FEniCS forum but is instead the Firedrake project. However, your question goes to UFL which is shared between the two projects. The UFL logarithm function is spelt `ln`. `log` is a function which logs a message to screen and/or a log file.
